diff --git a/Makefile b/Makefile
index 6f483c8..9a058ea 100644
--- a/Makefile
+++ b/Makefile
@@ -231,29 +231,29 @@ install: all
 	(cd ${DESTDIR}${INSTALL_LIB_TARG} ; \
 		ln -sf ${TARGLIB}.so.${MAJOR}.${MINOR} ${TARGLIB}.so.${MAJOR} ; \
 		ln -sf ${TARGLIB}.so.${MAJOR} ${TARGLIB}.so)
-	install -m 0644 -D psm2.h ${DESTDIR}/usr/include/psm2.h
-	install -m 0644 -D psm2_mq.h ${DESTDIR}/usr/include/psm2_mq.h
-	install -m 0644 -D psm2_am.h ${DESTDIR}/usr/include/psm2_am.h
+	install -m 0644 -D psm2.h ${DESTDIR}/include/psm2.h
+	install -m 0644 -D psm2_mq.h ${DESTDIR}/include/psm2_mq.h
+	install -m 0644 -D psm2_am.h ${DESTDIR}/include/psm2_am.h
 ifneq (fedora,${DISTRO})
 	install -m 0644 -D 40-psm.rules ${DESTDIR}$(UDEVDIR)/rules.d/40-psm.rules
 endif
 	# The following files and dirs were part of the noship rpm:
-	mkdir -p ${DESTDIR}/usr/include/hfi1diag
-	mkdir -p ${DESTDIR}/usr/include/hfi1diag/linux-x86_64
-	mkdir -p ${DESTDIR}/usr/include/hfi1diag/ptl_ips
-	install -m 0644 -D ptl_ips/ipserror.h ${DESTDIR}/usr/include/hfi1diag/ptl_ips/ipserror.h
-	install -m 0644 -D include/linux-x86_64/bit_ops.h ${DESTDIR}/usr/include/hfi1diag/linux-x86_64/bit_ops.h
-	install -m 0644 -D include/linux-x86_64/sysdep.h ${DESTDIR}/usr/include/hfi1diag/linux-x86_64/sysdep.h
-	install -m 0644 -D include/opa_udebug.h ${DESTDIR}/usr/include/hfi1diag/opa_udebug.h
-	install -m 0644 -D include/opa_debug.h ${DESTDIR}/usr/include/hfi1diag/opa_debug.h
-	install -m 0644 -D include/opa_intf.h ${DESTDIR}/usr/include/hfi1diag/opa_intf.h
-	install -m 0644 -D include/opa_user.h ${DESTDIR}/usr/include/hfi1diag/opa_user.h
-	install -m 0644 -D include/opa_service.h ${DESTDIR}/usr/include/hfi1diag/opa_service.h
-	install -m 0644 -D include/opa_common.h ${DESTDIR}/usr/include/hfi1diag/opa_common.h
-	install -m 0644 -D include/opa_byteorder.h ${DESTDIR}/usr/include/hfi1diag/opa_byteorder.h
-	install -m 0644 -D include/psm2_mock_testing.h ${DESTDIR}/usr/include/hfi1diag/psm2_mock_testing.h
-	install -m 0644 -D include/hfi1_deprecated.h ${DESTDIR}/usr/include/hfi1diag/hfi1_deprecated.h
-	install -m 0644 -D include/opa_revision.h ${DESTDIR}/usr/include/hfi1diag/opa_revision.h
+	mkdir -p ${DESTDIR}/include/hfi1diag
+	mkdir -p ${DESTDIR}/include/hfi1diag/linux-x86_64
+	mkdir -p ${DESTDIR}/include/hfi1diag/ptl_ips
+	install -m 0644 -D ptl_ips/ipserror.h ${DESTDIR}/include/hfi1diag/ptl_ips/ipserror.h
+	install -m 0644 -D include/linux-x86_64/bit_ops.h ${DESTDIR}/include/hfi1diag/linux-x86_64/bit_ops.h
+	install -m 0644 -D include/linux-x86_64/sysdep.h ${DESTDIR}/include/hfi1diag/linux-x86_64/sysdep.h
+	install -m 0644 -D include/opa_udebug.h ${DESTDIR}/include/hfi1diag/opa_udebug.h
+	install -m 0644 -D include/opa_debug.h ${DESTDIR}/include/hfi1diag/opa_debug.h
+	install -m 0644 -D include/opa_intf.h ${DESTDIR}/include/hfi1diag/opa_intf.h
+	install -m 0644 -D include/opa_user.h ${DESTDIR}/include/hfi1diag/opa_user.h
+	install -m 0644 -D include/opa_service.h ${DESTDIR}/include/hfi1diag/opa_service.h
+	install -m 0644 -D include/opa_common.h ${DESTDIR}/include/hfi1diag/opa_common.h
+	install -m 0644 -D include/opa_byteorder.h ${DESTDIR}/include/hfi1diag/opa_byteorder.h
+	install -m 0644 -D include/psm2_mock_testing.h ${DESTDIR}/include/hfi1diag/psm2_mock_testing.h
+	install -m 0644 -D include/hfi1_deprecated.h ${DESTDIR}/include/hfi1diag/hfi1_deprecated.h
+	install -m 0644 -D include/opa_revision.h ${DESTDIR}/include/hfi1diag/opa_revision.h
 
 specfile:
 	sed -e 's/@VERSION@/'${VERSION_RELEASE}'/g' ${RPM_NAME}.spec.in | \
diff --git a/buildflags.mak b/buildflags.mak
index 9cae815..1554539 100644
--- a/buildflags.mak
+++ b/buildflags.mak
@@ -61,7 +61,7 @@ export os ?= $(shell uname -s | tr '[A-Z]' '[a-z]')
 export arch := $(shell uname -p | sed -e 's,\(i[456]86\|athlon$$\),i386,')
 
 ifeq (${CCARCH},gcc)
-	export CC := gcc
+	export CC := ${CC}
 else
 	ifeq (${CCARCH},gcc4)
 		export CC := gcc4
@@ -89,7 +89,7 @@ BASECFLAGS += -DPSM2_MOCK_TESTING=1
 # we skip the linker script for testing version, we want all symbols to be
 # reachable from outside the library
 else
-LDFLAGS += -Wl,--version-script psm2_linker_script.map
+#LDFLAGS += -Wl,--version-script psm2_linker_script.map
 endif
 
 WERROR := -Werror
diff --git a/compat/buildflags.mak b/compat/buildflags.mak
index 7e4938d..b02b968 100644
--- a/compat/buildflags.mak
+++ b/compat/buildflags.mak
@@ -58,7 +58,7 @@ export arch := $(shell uname -p | sed -e 's,\(i[456]86\|athlon$$\),i386,')
 export CCARCH ?= gcc
 
 ifeq (${CCARCH},gcc)
-	export CC := gcc
+	export CC := ${CC}
 else
 	ifeq (${CCARCH},gcc4)
 		export CC := gcc4
