diff -ur '--exclude=.git' /reg/neh/home/cpo/git/cpsw/Framework/config.mak /reg/neh/home/cpo/junk/cpsw/config.mak
--- /reg/neh/home/cpo/git/cpsw/Framework/config.mak	2017-11-16 15:13:51.681141804 -0800
+++ /reg/neh/home/cpo/junk/cpsw/config.mak	2017-11-16 17:49:12.536175251 -0800
@@ -15,20 +15,24 @@
 # arbitrarily but should not characters besides
 # [0-9][a-z][A-Z][.][-][_]
 
-PACKAGE_TOP=/afs/slac/g/lcls/package
+#PACKAGE_TOP=/afs/slac/g/lcls/package
+PACKAGE_TOP=/reg/common/package
+#PACKAGE_TOP=${CONDA_PREFIX}
 
 # By default the 'host' architecture is built. But
 # you can redefine the name of the host architecture:
 # HARCH = www
 #HARCH=linux-x86_64
-HARCH=rhel6-x86_64
+#HARCH=rhel6-x86_64
+#HARCH=x86_64-rhel7-gcc48-opt
+HARCH=x86_64-rhel7-gcc72-opt
 
 # Assuming you also want 'xxx' and 'yyy' then you'd
 # add these to the 'ARCHES' variable:
 
 # ARCHES += xxx yyy
 
-ARCHES += buildroot-2016.11.1-x86_64 buildroot-2015.02-x86_64
+# ARCHES += buildroot-2016.11.1-x86_64 buildroot-2015.02-x86_64
 
 
 # Next, you need to define prefixes (which may include
@@ -40,6 +44,8 @@
 #
 # (you can override 'gcc' by another compiler by setting CC_<arch>)
 
+CC_x86_64-rhel7-gcc72-opt = $(CC)
+
 # definitions for multiple buildroot targets
 
 BR_HARCH=linux-x86_64
@@ -94,8 +100,10 @@
 #
 
 # 
-BOOST_VERSION=1.63.0
+#BOOST_VERSION=1.63.0
+BOOST_VERSION=1.57.0
 BOOST_PATH=$(PACKAGE_TOP)/boost/$(BOOST_VERSION)
+#BOOST_PATH=$(PACKAGE_TOP)
 
 boost_DIR_default=$(BOOST_PATH)/$(TARCH)
 
@@ -107,6 +115,8 @@
 # 
 YAML_CPP_VERSION         = yaml-cpp-0.5.3
 YAML_CPP_PATH            = $(PACKAGE_TOP)/yaml-cpp/$(YAML_CPP_VERSION)
+#YAML_CPP_PATH            = $(PACKAGE_TOP)
+#YAML_CPP_PATH            = /reg/g/pcds/package/external/yaml-cpp
 
 yaml_cpp_DIR_default=$(YAML_CPP_PATH)/$(TARCH)
 
diff -ur '--exclude=.git' /reg/neh/home/cpo/git/cpsw/Framework/src/cpsw_entry.h /reg/neh/home/cpo/junk/cpsw/src/cpsw_entry.h
--- /reg/neh/home/cpo/git/cpsw/Framework/src/cpsw_entry.h	2017-11-16 15:13:51.738143291 -0800
+++ /reg/neh/home/cpo/junk/cpsw/src/cpsw_entry.h	2017-11-16 15:53:43.454513051 -0800
@@ -42,7 +42,7 @@
 		static const int CONFIG_PRIO_OFF      = 0;
 		static const int DFLT_CONFIG_PRIO     = CONFIG_PRIO_OFF;
 		static const int DFLT_CONFIG_PRIO_DEV = 1; // on for containers
-		static const double DFLT_POLL_SECS    = 0.; 
+		static constexpr double DFLT_POLL_SECS    = 0.; 
 
 		class                              CUniqueHandle;
 		typedef shared_ptr<CUniqueHandle>   UniqueHandle;
diff -ur '--exclude=.git' /reg/neh/home/cpo/git/cpsw/Framework/src/cpsw_path.cc /reg/neh/home/cpo/junk/cpsw/src/cpsw_path.cc
--- /reg/neh/home/cpo/git/cpsw/Framework/src/cpsw_path.cc	2017-11-16 15:13:51.809145143 -0800
+++ /reg/neh/home/cpo/junk/cpsw/src/cpsw_path.cc	2017-11-16 15:53:43.602516910 -0800
@@ -459,7 +459,7 @@
 {
 Address      found;
 ConstDevImpl h;
-Path         rval = make_shared<CPathImpl>( *this );
+ Path         rval = boost::make_shared<CPathImpl>( *this );
 CPathImpl    *p    = _toPathImpl( rval );
 const char  *sl;
 
@@ -630,7 +630,7 @@
 
 PathImpl CPathImpl::cloneAsPathImpl() const
 {
-	return make_shared<CPathImpl>( *this );
+  return boost::make_shared<CPathImpl>( *this );
 }
 
 Path CPathImpl::concat(ConstPath p) const
diff -ur '--exclude=.git' /reg/neh/home/cpo/git/cpsw/Framework/src/cpsw_proto_mod_rssi.cc /reg/neh/home/cpo/junk/cpsw/src/cpsw_proto_mod_rssi.cc
--- /reg/neh/home/cpo/git/cpsw/Framework/src/cpsw_proto_mod_rssi.cc	2017-11-16 15:13:51.838145899 -0800
+++ /reg/neh/home/cpo/junk/cpsw/src/cpsw_proto_mod_rssi.cc	2017-11-16 15:53:43.573516154 -0800
@@ -40,7 +40,7 @@
 		return inpQ_->push( bc, timeout );
 	} else {
 		CTimeout abst( inpQ_->getAbsTimeoutPush( timeout ) );
-		return inpQ_->pop( &abst );
+		return inpQ_->pop( &abst )!=0;
 	}	
 }
 
diff -ur '--exclude=.git' /reg/neh/home/cpo/git/cpsw/Framework/src/defs.mak /reg/neh/home/cpo/junk/cpsw/src/defs.mak
--- /reg/neh/home/cpo/git/cpsw/Framework/src/defs.mak	2017-11-16 15:13:51.944148664 -0800
+++ /reg/neh/home/cpo/junk/cpsw/src/defs.mak	2017-11-16 17:41:23.676950998 -0800
@@ -20,6 +20,9 @@
 AR_host    :=$(AR)
 RANLIB_host:=$(or $(RANLIB),ranlib)
 
+CC_default  :=$(CC_HOST)
+CXX_default :=$(CXX_host)
+
 GIT_VERSION_STRING:=$(shell echo -n '"'`git describe --always --dirty`'"')
 
 # if CROSS_xxx (for target architecture 'xxx' is not defined then try CROSS_default)
@@ -71,7 +74,8 @@
 CROSS_default=$(addsuffix -,$(filter-out $(HARCH),$(TARCH)))
 
 # Tools
-CC     =$(CROSS)$(or $(CC_$(TARNM)),$(CC_default),gcc)
+#CC     =$(CROSS)$(or $(CC_$(TARNM)),$(CC_default),gcc)
+#CC     =$(CC)
 CXX    =$(CROSS)$(or $(CXX_$(TARNM)),$(CXX_default),g++)
 AR     =$(CROSS)$(or $(AR_$(TARNM)),$(AR_default),ar)
 RANLIB =$(CROSS)$(or $(RANLIB_$(TARNM)),$(RANLIB_default),ranlib)
Only in /reg/neh/home/cpo/junk/cpsw/src/doc: INSTALL.html
Only in /reg/neh/home/cpo/junk/cpsw/src/doc: README.configData.html
Only in /reg/neh/home/cpo/junk/cpsw/src/doc: README.yamlDefinition
Only in /reg/neh/home/cpo/junk/cpsw/src/doc: README.yamlDefinition.html
Only in /reg/neh/home/cpo/junk/cpsw/src: git_version_string.h
diff -ur '--exclude=.git' /reg/neh/home/cpo/git/cpsw/Framework/src/makefile /reg/neh/home/cpo/junk/cpsw/src/makefile
--- /reg/neh/home/cpo/git/cpsw/Framework/src/makefile	2017-11-16 15:13:51.955148950 -0800
+++ /reg/neh/home/cpo/junk/cpsw/src/makefile	2017-11-16 17:47:30.431513138 -0800
@@ -11,7 +11,7 @@
 
 include $(CPSW_DIR)/defs.mak
 
-POSTBUILD_SUBDIRS=test
+#POSTBUILD_SUBDIRS=test
 
 HEADERS = cpsw_api_user.h cpsw_api_builder.h cpsw_api_timeout.h cpsw_error.h
 DOCS = doc/README.yamlDefinition doc/README.yamlDefinition.html doc/README.configData.html doc/INSTALL.html
@@ -117,5 +117,5 @@
 %.html: %
 	sed -e 's/&/\&amp/g' -e 's/</\&lt/g' -e 's/>/\&gt/g' $< | sed -e '/THECONTENT/{r /dev/stdin' -e 'd}' tmpl.html > $@
 
-test: sub-./test@test
-	@true
+#test: sub-./test@test
+#	@false
Only in /reg/neh/home/cpo/junk/cpsw/src: O.buildroot-2016.11.1-x86_64
Only in /reg/neh/home/cpo/junk/cpsw/src: O.x86_64-rhel7-gcc72-opt
diff -ur '--exclude=.git' /reg/neh/home/cpo/git/cpsw/Framework/src/rules.mak /reg/neh/home/cpo/junk/cpsw/src/rules.mak
--- /reg/neh/home/cpo/git/cpsw/Framework/src/rules.mak	2017-11-16 15:13:51.963149159 -0800
+++ /reg/neh/home/cpo/junk/cpsw/src/rules.mak	2017-11-16 17:19:52.609289511 -0800
@@ -247,6 +247,11 @@
 
 build: $(TGTS)
 
+print:
+	@echo "HELLO CPO"
+	@echo CC $(CC)
+	@echo CXX $(CXX)
+
 run_tests: $(addsuffix _run,$(FILTERED_TBINS))
 	@echo "ALL TESTS PASSED"
 
