diff --git a/buildflags.mak b/buildflags.mak
index 34fdf1c..4962c9e 100644
--- a/buildflags.mak
+++ b/buildflags.mak
@@ -68,7 +68,7 @@ ifeq (1,$(PSM_HAVE_SCIF))
 	SCIF_LINK_FLAGS += -lscif
 endif
 
-WERROR := -Werror
+WERROR := -Wno-error
 INCLUDES := -I. -I$(top_srcdir)/include -I$(top_srcdir)/mpspawn \
 	-I$(top_srcdir)/include/$(os)-$(arch) $(SCIF_INCLUDE_FLAGS)
 BASECFLAGS += $(BASE_FLAGS) $(if $(MIC:0=),$(if $(filter $(CC),icc),-mmic,-D__MIC__)) \
